Kudu.Core/SourceControl/Git/LibGit2SharpRepository.cs

```diff
@@ -0,0 +1,396 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Kudu.Contracts.Settings;
+using Kudu.Core.Tracing;
+using System.IO;
+using Kudu.Core.Infrastructure;
+using LibGit2Sharp;
+using Kudu.Contracts.Tracing;
+using System.Text.RegularExpressions;
+
+namespace Kudu.Core.SourceControl.Git
+{
+    public class LibGit2SharpRepository : IRepository
+    {
+        private const string _remoteAlias = "external";
+        private readonly ITraceFactory _tracerFactory;
+        private readonly IDeploymentSettingsManager _settings;
+        private readonly GitExeRepository _legacyGitExeRepository;
+
+        public LibGit2SharpRepository(IEnvironment environment, IDeploymentSettingsManager settings, ITraceFactory tracerFactory)
+        {
+            _tracerFactory = tracerFactory;
+            _settings = settings;
+            RepositoryPath = environment.RepositoryPath;
+            _legacyGitExeRepository = new GitExeRepository(environment, settings, tracerFactory);
+        }
+
+        public string CurrentId
+        {
+            get 
+            {
+                using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
```

Comment:
how expensive is it for creating and disposing repo object per api as opposed to create one and hold on to it in ctor?    If nothing, then it is great.  Otherwise, we might consider.   Given all git calls are synchronized (we took care of that) - no concurrent issue.    Also the physical repo is stateful, object state should not be an issue as well.   Thought?